### ISSUE TITLE
Code with pipe character in table is impossible to write correctly

### ISSUE THREAD

**Issue by iFreilicht** [rust-lang/mdBook#2000]
### Problem

This was discovered in [NixOS/nix#7770](https://github.com/NixOS/nix/pull/7700)

We have a table with all operators, but `||` doesn't get rendered correctly:

![image](https://user-images.githubusercontent.com/9742635/215285880-344e3cdb-3ce7-4695-84a1-0987d2ab9b2f.png)

I thought that maybe removing the `\` from the markdown would solve this, but no. Comparing a few types of spans:

```markdown
| code | bold   | ital |
| ---- | ------ | ---- |
|  `|` |  __|__ |  *|* |
| `\|` | __\|__ | *\|* |
```

![Screenshot 2023-01-28 at 20 06 06](https://user-images.githubusercontent.com/9742635/215286308-ebb1d322-0142-4de1-b232-c41519d04bac.png)

The [documentation claims that tables are implemented](https://rust-lang.github.io/mdBook/format/markdown.html?highlight=tables#tables) according to the [GFM tables extension](https://github.github.com/gfm/#tables-extension-).

However, this is not the case. From the [GFM spec, Example 200](https://github.github.com/gfm/#example-200):

> Include a pipe in a cell’s content by escaping it, including inside other inline spans:
> ```
> | f\|oo  |
> | ------ |
> | b `\|` az |
> | b **\|** im |
> ```

### Steps

1. Create a new mdbook
2. Paste this code into `chapter_1.md`:
    ```markdown
    | code | bold   | ital |
    | ---- | ------ | ---- |
    |  `|` |  __|__ |  *|* |
    | `\|` | __\|__ | *\|* |
    ```
3. Serve or build the mdbook and observe that either way of putting a pipe symbol inside a code span inside a table doesn't work:
    ![Screenshot 2023-01-28 at 20 06 06](https://user-images.githubusercontent.com/9742635/215286308-ebb1d322-0142-4de1-b232-c41519d04bac.png)

### Possible Solution(s)

It seems mdbook treats the code span as a self-contained unit, so `\|` does not have special meaning inside the table as it should.

While IMO it would be more intuitive to just have the non-escaped pipe symbol work inside code spans, adhering to the GFM spec would be best in terms of compatibility, so that's what I'd suggest. 

### Notes

In the end, this minimal markdown

```md
| | |
| - | - |
| `\|` | |
```

should produce this HTML

```html
<table>
  <thead>
    <tr>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><code class="hljs">|</code></td>
      <td></td>
    </tr>
  </tbody>
</table>
```

While right now it produces this:

```html
<table>
  <thead>
    <tr>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><code class="hljs">\|</code></td>
      <td></td>
    </tr>
  </tbody>
</table>
```

### Version

```text
mdbook v0.4.25
```

**Comment by iFreilicht** [comment]
I did a bit more research and found out this is actually a known problem in pulldown-cmark since 2019, see [pulldown-cmark#356](https://github.com/raphlinus/pulldown-cmark/issues/356). Not sure if this should stay open since it's a problem in mdbook that requires updating a dependency (once the bug is fixed there).

**Comment by ehuss** [comment]
It probably doesn't hurt to keep this open, but yea it is blocked on https://github.com/raphlinus/pulldown-cmark/issues/356.

**Comment by schungx** [comment]
Yes, this has always been a problem.  It is impossible to put a `` `\` `` inside a table.

The work around is to use `<code></code>` tags.

```text
| <code>This is `\\`</code>|
```
